spa/src/components/paymentProviders/stripe/stripeFns.js

```
@@ -105,3 +106,48 @@ export async function createPaymentMethod(stripe, card, data) {
     billing_details
   });
 }
+
+export function getPaymentSuccessUrl(
+  baseUrl,
```

How about we destructure these args like this:

```js
export function getPaymentSuccessUrl({ thankYouRedirectUrl, amount, emailHash, etc etc })
```

Reason I suggest it is because it would be easy to accidentally put the args in the wrong order, and it will more-or-less silently fail, but object prop order doesn't matter.

In other words...

```js
// not obviously a bug
const return_url = getPaymentSuccessUrl(
      window.location.origin,
      thank_you_redirect || '',
      emailHash,
      amount,
      getFrequencyThankYouText(frequency),
      contributorEmail,
      pageSlug,
      rpSlug,
      pathname,
      stripeClientSecret
    );

// works fine even though we flipped email and amount
const return_url = getPaymentSuccessUrl({
  baseUrl: window.location.url,
  thankYouRedirect: thank_you_redirect || '',
  emailHash,
  amount,
  frequencyDisplayValue: getFrequencyThankYouText(frequency),
  contributorEmail,
  pageSlug,
  rpSlug,
  pathname,
  stripeClientSecret
});
```

---

Had same thought. You busted me being lazy.

8302702706625c